### PR TITLE
Basic logger system

### PR DESCRIPTION
**Pull request by menion**
Basic system for logger system for shared libraries.

Logger is based on the existing light version in Locus API & experience with own logger system on the Android.

**Optimal usage**
1. set own implementation of `ILogger` interface with `Logger.registerLogger`.
2. use static methods of `Logger.d`, `Logger.w` etc.

**Interesting info**
- not necessary to create something like `Logger.getInstance`
- not necessary to supply "TAG" (optional). It may be auto-generated
- better Kotlin support

Check it and compare it to the logger style you use in your own project. The main usage of this system should be in shared libraries. There is no dependency, no hidden requirements. So it is very easy to use everywhere ...

### CODE REVIEW

**Review by hrstka — APPROVED** [review]

**Review by radim-asamm — APPROVED** [review]
Thanks for introducing me to an interesting way how to use Throwable! I see no reason why this solution shouldn't work. It has been working for years, I suppose. Anyway: this job looks to me like a typical job for some lightweight DI Android FW. I understand introducing it is not justified by logging purposes only. But, for the future sake: Would Locus Api users benefit from something like Koin? Or is introducing it just an unnecessary overengineering?

**Review by voldapet — APPROVED** [review]

File: src/main/java/com/asamm/logger/Logger.kt [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+    /**
+     * Register [logger] implementation.
+     */
+    fun registerLogger(logger: ILogger) {
+        Logger.logger = logger
+    }
+
+    interface ILogger {
+
+        fun logV(tag: String, msg: String, vararg args: Any)
```
Comment (janaton) [inline src/main/java/com/asamm/logger/Logger.kt:186]: [question] What is the format of args placeholder in text? This is important and I believe should be put in comment.

File: README.md [hunk 2]
```diff
@@ -1,2 +1,16 @@
...
 Logger for Asamm libraries and tools.
+
+## Quick start
+
+Add dependency to your `build.gradle` module config
+
+```gradle
+dependencies {
+     // get Logger
+     compile 'com.asamm:logger-asamm:[latest]'
```
Comment (janaton) [inline README.md]: [suggestion] `logger-asamm` -> `asamm-logger`

**Review by janaton — APPROVED** [review]

File: src/main/java/com/asamm/logger/Logger.kt (hunk 1 above)
Comment (menion) [reply src/main/java/com/asamm/logger/Logger.kt:186]: Thanks for asking. I've added the necessary info into Javadoc.
This highly depends on clients that will log messages, but good practice will be to follow 'String.format' system.

File: README.md (hunk 2 above)
Comment (menion) [reply README.md]: 😃 , oki, I'll name this dependency just a logger.

**Comment by menion** [comment]
Thanks guys. PR merged, I'm going to prepare CI/CD system.

**Pull request merged**